### ISSUE TITLE
[FR] Do not hardcode `/config`, `/downloads`, ...

### ISSUE THREAD

**Issue by charludo** [kieraneglin/pinchflat#490]
**Is your feature request related to a problem? Please describe.**
This would allow the project to be run on various linux distros, without docker (for example, I am currently trying to package this project for nixpkgs).

**Describe the solution you'd like**
Read `CONFIG_DIRECTORY`, `DOWNLOADS_DIRECTORY`, ... from the environment, similar to the other environment variables already in use.

**Additional context**
While mapping host paths to `/config` and `/downloads` is certainly convenient in docker-based setups, this currently makes it very hard to run in a non-docker setup.

**Comment by kieraneglin** [comment]
Hey there! Thanks for the suggestion (:

I'm hesitant to do this since non-docker installation isn't supported whatsoever. While it's all technically possible, I don't have the capacity to provide troubleshooting and support for all the types of issues that come from a non-standardized installation method. I totally get that the change you're requesting is simple in isolation, but as a one-man show I need to ruthlessly guard my time and energy so that I can focus on bugfixes and new features instead of getting bogged down by support requests. On top of that, I don't _personally_ see the benefit of adding Pinchflat to a non-Docker installation repository when compared to the ease of just using Docker.

That said, I know only the basic idea around Nix and it's package management system, so I may be missing something. Is there some benefit that I'm not understanding? Please tell me more about the use case!

**Comment by charludo** [comment]
Hi, thanks for the quick response!

 I absolutely get not wanting to support systems and usecases you don't personally use. That is completely legitimate.

The only thing I can (try to 😄) convince you of is that installation through nixpkgs would definitely be standardized and convenient. Theoretically, users would only need to add
```nix
services.pinchflat.enable = true;
```
to their config to build, install, configure, and (auto)start Pinchflat. Though to be fair, the more realistic usecase would probably look like
```nix
services.pinchflat = {
    enable = true;
    secretsFile = config.sops.secrets.pinchflat.path;
    mediaDir = "/path/to/media/dir"; # 😃 
};
```

For people already using nix for everything, I would definitely argue this is the more convenient way of installing Pinchflat 😄 

Would you be open to merging a PR which enables configuring the dirs? If so, I'll try my hand at it. Otherwise, if you are certain you do not want to enable these config options here (which again, is totally fine and understandable), I would create a patchfile which patches the changes in during the nix build process. Though that would be more prone to errors and harder to keep updated, so I personally hope for the former option 😁

**Comment by kieraneglin** [comment]
After thinking on it some more I'm open to adding it to the app. But if I get an influx of support requests, I may just have to tell them I don't have the capacity to help troubleshoot all these things.

Is there a way for you to pull from a specific branch while this package is being created? That way I can consolidate any required changes into a single branch until you've confirmed it's ready to roll

**Comment by kieraneglin** [comment]
Resolved by #493
